bpfd-operator/internal/bpfd-client-helpers.go

```diff
@@ -382,3 +400,31 @@ func PrintNodeState(state map[ProgramKey]ExistingReq) {
 		log.V(1).Info("Node State---->", "ProgramKey", k, "Value", v)
 	}
 }
+
+func ConvertToBpfProgramConfigList(BpfProgramConfig *bpfdiov1alpha1.BpfProgramConfig, NodeIface string) (nodeState []*bpfdiov1alpha1.BpfProgramConfig) {
```

Comment:
convert this to take in a bpfProgramConfig.Spec

Reply:
I think you meant for `BuildBpfdLoadRequest()` to take in a `bpfProgramConfig.Spec` struct and for `ConvertToBpfProgramConfigList()` to return a list of `bpfProgramConfig.Spec` instead of a list of ``bpfProgramConfig` structs.

Done on next commit.